deeppavlov/skills/dsl_skill/dsl_skill.py

```diff
@@ -0,0 +1,188 @@
+        return (*list(map(list, zip(*starmap(cls.handle, zip_longest(utterances_batch, history_batch, states_batch))))),)
+
+    @staticmethod
+    def __add_to_collection(cls: 'DSLMeta'):
+        """
+        Adds Skill class to Skill classes collection
+        Args:
+            cls: Skill class
+        """
+        DSLMeta.skill_collection[cls.name] = cls
+
+    @staticmethod
+    def __handle(cls: 'DSLMeta',
+                 utterance: str,
+                 history: str,
+                 state: str) -> ResponseType:
+        """
+        Handles what is going to be after a message from user arrived.
+        Simple usage:
+        ExampleSkill.handle(Request(<message>, <user_id>))
+
+        Args:
+            cls: instance of callee's class
+            utterance: a message to be handled
+            history: history of dialog
+            state: state
+        Returns:
+            result: handler function's result if succeeded
+        """
+        current_handler = cls.__select_handler(utterance, history, state)
+        return cls.__run_handler(current_handler, utterance, history, state)
+
+    def __select_handler(cls, message: str,
+                         history: str,
+                         state: str) -> Optional[Callable]:
+        """
+        Selects handler with the highest priority that could be triggered from the passed state and message.
+        Returns:
+             handler function that is selected and None if no handler fits request
+        """
+        available_handlers = cls.state_to_handler[state]
+        available_handlers.extend(cls.universal_handlers)
+        available_handlers.sort(key=lambda h: h.priority, reverse=True)
+        for handler in available_handlers:
+            if handler.check(message, history):
+                return handler.func
+
+    def __run_handler(cls, handler: Optional[Callable],
+                      message: str,
+                      history: str,
+                      state: str) -> ResponseType:
+        """
+        Runs specified handler for current message and context
+        Args:
+            handler: handler to be run. If None, on_invalid_command is returned
+        Returns:
+             ResponseType
+        """
+        if handler is None:
+            return ResponseType(cls.on_invalid_command, cls.null_confidence, None)
+        try:
+            return ResponseType(*handler(message, history, state))
+        except Exception as exc:
+            return ResponseType(str(exc), 1.0, None)
+
+    @staticmethod
+    def handler(commands: List[str] = None,
+                state: str = None,
+                context_condition=None,
+                priority: int = 0):
+        """
+        Decorator to be used in skills' classes.
+        Sample usage:
+        class ExampleSkill(metaclass=ZDialog):
+            @ZDialog.handler(commands=["hello", "hey"], state="greeting")
+            def __greeting(message: str):
+                ...
```

Comment:
Documentation build fails here.
You can try it by running `make html` from the [docs](https://github.com/eora-dialog-systems/DeepPavlov/tree/master/docs) directory